app/models/balance.rb

```diff
@@ -49,6 +44,70 @@ def build_wrapper(other_user)
 
   attr_reader :user
 
+  # rubocop:disable Metrics/AbcSize
+  def payment_debts
+    Payment
+      .where(
+        "(user_id = #{user.id} OR payer_id = #{user.id}) " \
+        "AND user_id != payer_id"
+      )
+      .group(
+        "CASE WHEN user_id = #{user.id} THEN payer_id " \
+        "WHEN payer_id = #{user.id} THEN user_id END"
+      )
+      .having(
+        "SUM(CASE WHEN user_id = #{user.id} THEN -balance_cents " \
+        "WHEN payer_id = #{user.id} THEN balance_cents END) < 0"
+      )
+      .pluck(
+        "CASE WHEN user_id = #{user.id} THEN payer_id " \
+        "WHEN payer_id = #{user.id} THEN user_id END, " \
+        "SUM(CASE WHEN user_id = #{user.id} THEN -balance_cents " \
+        "WHEN payer_id = #{user.id} THEN balance_cents END)"
+      ).to_h
+  end
```

Comment:
What I don't like here is that it's almost identical to `payment_balances`. You could build a query object called `PaymentQuery` with chainable methods for this to avoid this redundancy. It would look like:
```
class PaymentQuery
  attr_reader :query
  
  def initialize(user)
    @user = user
  end

  def total_debt
    all_users.grouped_by_id.having_sum.pluck_balance.to_h
  end

  def total_balance
    all_users.grouped_by_id.pluck_balance.to_h
  end

  private

  def all_users
    @query = Payment
      .where(
        "(user_id = #{user.id} OR payer_id = #{user.id}) " \
        "AND user_id != payer_id"
      )
  end

  def grouped_by_id
    @query = query.group(
        "CASE WHEN user_id = #{user.id} THEN payer_id " \
        "WHEN payer_id = #{user.id} THEN user_id END"
      )
  end

  def having_sum
    @query = query.having(
        "SUM(CASE WHEN user_id = #{user.id} THEN -balance_cents " \
        "WHEN payer_id = #{user.id} THEN balance_cents END) < 0"
      )
  end

  def pluck_balance
    @query = query.pluck(
        "CASE WHEN user_id = #{user.id} THEN payer_id " \
        "WHEN payer_id = #{user.id} THEN user_id END, " \
        "SUM(CASE WHEN user_id = #{user.id} THEN -balance_cents " \
        "WHEN payer_id = #{user.id} THEN balance_cents END)"
      )
  end
end

```

Of course this doesn't have to be a separate object, it's just an example of how to DRY out such queries,